src/Router/UrlGenerator.php

```diff
@@ -47,40 +62,120 @@ public static function buildUrl($url, $replace = [], $flags = HTTP_URL_REPLACE,
                    | HTTP_URL_STRIP_PASS;
         }
 
-        // Filter $url and $replace arrays to strip out unknown segments
-        array_change_key_case($url, CASE_LOWER);
-        array_change_key_case($replace, CASE_LOWER);
+        // Parse input
+        if (is_string($url)) {
+            $url = parse_url($url);
+        }
+        if (is_string($replace)) {
+            $replace = parse_url($replace);
+        }
+
+        // Prepare input data
+        $cleanUrlArray = function (&$url) use ($urlSegments) {
+            if (!is_array($url)) {
```

Comment:
I don't think this check would be needed. At the point that this function is used, both `$url` and `$replace` would have to be arrays.

Reply:
Nope, if something that isn't a string or an array is passed, then it's not an array. I.e. false or null would be the main offenders.

Reply:
@LukeTowers if `false` or `null` are accepted, then the docblock will need to be updated, as it only says `array|string` at the moment.

Reply:
Done